src/sage/libs/gap/util.pyx

```diff
@@ -434,6 +443,18 @@ cdef Obj gap_eval(str gap_string) except? NULL:
 ### Error handler ##########################################################
 ############################################################################
 
+
+cdef object keyboard_interrupt_exc = KeyboardInterrupt(), alarm_interrupt_exc = KeyboardInterrupt("alarm interrupt")
+
+
+try:
+    from cysignals.signals import AlarmInterrupt
+except ImportError:
+    pass
```

Comment:
The indirection won't really help if the doctest runner can only catch `AlarmInterrupt`, you might as well just add cysignals back as a dependency and simplify things. If someone wants to split off our libgap interface in the future they can figure out what to do.

Reply:
it doesn't need to be a build time dependency, just a run time dependency.

Originally I was just thinking of a dependency injection thing where when sage import it it sets the alarm interrupt object, but I figure it isn't that useful.

Reply:
thinking about it there has been an issue in https://github.com/sagemath/sage/issues/39829 , possibly related to some such optimization.